### PR TITLE
[RG-138] Introduce --project cmdline option to open a project directly at startup

### PR DESCRIPTION
**Pull request by TarasSydor**
> ### Motivate of the pull request
This PR fulfills RG-138 and introduces new '--project' cmdline option.

> ### Describe the technical details
Command line arguments parsing has been extended with '--project' option. In case it's used together with '--script', an error is given and startup fails.
>
> #### What does this pull request change?
'--project' allows setting a project file (*.ospr) in the cmdline and corresponding gets opened after startup. As if the application was started without any cmdline options and Open Project action for indicated project was executed.

Tested on FOEDAG and Raptor:
image

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [+] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

### CODE REVIEW

File: src/MainWindow/Session.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "Compiler/TaskManager.h"
 #include "Main/TclSimpleParser.h"
-#include "TopLevelInterface.h"
+#include "main_window.h"
```
Comment (KochynVolodymyr) [inline src/MainWindow/Session.cpp:26]: revert this

File: src/MainWindow/TopLevelInterface.h [hunk 2]
```diff
@@ -20,12 +20,15 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 
+#include <string>
+
 namespace FOEDAG {
 
 class TopLevelInterface {
  public:
   virtual ~TopLevelInterface() = default;
   virtual void gui_start(bool showWP) = 0;
+  virtual void openProject(const std::string& projectFile) = 0;
```
Comment (KochynVolodymyr) [inline src/MainWindow/TopLevelInterface.h:31]: lets use QString

File: src/MainWindow/main_window.h [hunk 3]
```diff
@@ -107,6 +107,7 @@ class MainWindow : public QMainWindow, public TopLevelInterface {
 
   void cleanUpDockWidgets(std::vector<QDockWidget*>& dockWidgets);
   void openProject(const QString& project);
+  void openProject(const std::string& projectFile) override;
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.h:110]: if use QString you don't need additional method

Comment (TarasSydor) [reply src/MainWindow/main_window.h]: agree. fixed

**Review by KochynVolodymyr — APPROVED** [review]

**Comment by alain-rs** [comment]
@TarasSydor please update or rebase, this is out of date

**Comment by alain-rs** [comment]
Please update the help too.

**Comment by TarasSydor** [comment]
@ravic-rs Tests on Raptor were successful: --project cmdline option and open_project tcl command opens desired project.
I noticed however one issues and couldn't immediately solve it. It's both on Foedag and Raptor - if GUI was started with --project, it won't get properly closed. there are some dangling widgets.
I'll investigate that problem before merging it.

**Comment by TarasSydor** [comment]
> @ravic-rs Tests on Raptor were successful: --project cmdline option and open_project tcl command opens desired project. I noticed however one issues and couldn't immediately solve it. It's both on Foedag and Raptor - if GUI was started with --project, it won't get properly closed. there are some dangling widgets. I'll investigate that problem before merging it.

Issue solved by most recent commit.

**Review by ravic-rs — APPROVED** [review]
Looks good.

**Pull request merged**